common/elasticsearch/client_v7.go

```diff
@@ -221,73 +166,46 @@ func (c *elasticV7) ScanByQuery(ctx context.Context, request *ScanByQueryRequest
 	if err != nil {
 		return nil, err
 	}
-
-	var searchResult *elastic.SearchResult
-	var scrollService *elastic.ScrollService
-
-	if len(token.ScrollID) == 0 { // first call
-		searchResult, scrollService, err = c.scrollFirstPage(ctx, request.Index, request.Query)
-	} else {
-		searchResult, scrollService, err = c.scroll(ctx, token.ScrollID)
-	}
+	searchResult, err := c.scroll(ctx, request.Index, request.Query, token.ScrollID)
 
 	isLastPage := false
 	if err == io.EOF { // no more result
 		isLastPage = true
-		if scrollService != nil {
-			err := scrollService.Clear(ctx)
-			if err != nil {
-				c.logger.Warn("scrollService Clear fail", tag.Error(err))
-			}
+		if err := c.clearScroll(ctx, token.ScrollID); err != nil {
```

Comment:
the input should be `searchResult.ScrollID`

Reply:
good catch, thanks